charts/zookeeper/templates/role.yaml

```diff
@@ -0,0 +1,21 @@
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: {{ default "zookeeper" .Values.pod.serviceAccountName }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+{{ include "zookeeper.commonLabels" . | indent 4 }}
+rules:
+- apiGroups:
+  - zookeeper.pravega.io
+  resources:
+  - "*"
```

Comment:
shall we specify resources instead of *

Reply:
I  am not sure what resources we can give here. In our other operators as well, we have given `*`

Reply:
keep investigating what resources are needed exactly. we can move forward with current PR.

Reply:
It seems valid to me to be `*` since these are Zookeeper CRD resources which the Zookeeper operator is responsible for.

Reply:
Why is the service account that is used for the zk cluster pods given access to zookeeper operator crd resources?

If this role was given to the zookeeper-operator service account I'd understand but... actually it already has what it needs: https://github.com/pravega/zookeeper-operator/blob/master/charts/zookeeper-operator/templates/clusterrole.yaml

I thought the point of this PR was to give the ability for zk pods to run a non default service account (which is achieved as part of the changes in zookeepercluster_types.go ); but I don't understand why this service account needs to be given any role.   Think of the roles the "default" service account has when it runs the zk pods before this PR.  None (explicitly).

Reply:
So in summary: I understand and agree with all the changes in this PR except I don't understand any of the changes in charts/zookeeper/templates, except for service_account.yaml and zookeeper.yaml

Reply:
Additionally, I'd suggest that while the service account name should come from values.yaml, really the zk operator would actually provision the Service account resource itself (if name is provided).

Reply:
Thanks @sarlaccpit for your review. I have addressed your comments in another PR https://github.com/pravega/zookeeper-operator/pull/256